src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Float.cs

```diff
@@ -107,14 +108,16 @@ public static bool TryFormat(float value, Span<byte> destination, out int bytesW
                 return false;
             }
 
-            for (int i = 0; i < utf16Text.Length; i++)
+            try
             {
-                Debug.Assert(utf16Text[i] < 128, "A culture-invariant ToString() of a floating point expected to produce ASCII characters only.");
-                destination[i] = (byte)utf16Text[i];
+                bytesWritten = Encoding.UTF8.GetBytes(utf16Text, destination);
+                return true;
```

Comment:
Have you done any measurements to see if / by how much this regresses throughput for the ASCII case?

I'm wondering if there'd be any benefit to doing the original loop, but instead of an assert, falling back to using Encoding.UTF.GetBytes from that point on in the span.  I realize UTF8Encoding does something similar, so maybe it's not worth it.  It'd be good to understand what if any hit we're taking here, though.

Reply:
No. I haven't done any measurements here yet.

> I'm wondering if there'd be any benefit to doing the original loop, but instead of an assert, falling back to using Encoding.UTF8.GetBytes from that point on in the span

If there is any perf hit, I would rather try to make `UTF8Encoding.GetBytes` faster, as that is the overall more common usage scenario and would benefit anyone calling it (otherwise, we may just end up with several duplicated "fast paths" trying all ASCII and then falling back to `UTF8Encoding.GetBytes`)

Reply:
> No. I haven't done any measurements here yet.

Mind checking?

Reply:
Yes, working on that already (thought I mentioned that above, but looks like I didn't)

Reply:
I wasn't able to detect any real perf difference. For very small strings (like `5E-324`) there is some minor overhead of the call, but this is ~1ms over 65k invocations.

For larger strings (like the result of "G99"), the result is minimally faster (again, ~1ms over 65k invocations).

Reply:
Thanks for checking.

> For very small strings (like 5E-324) there is some minor overhead of the call, but this is ~1ms over 65k invocations.

How long did the 65K invocations take?